simple/stats/nodes.py

```diff
@@ -0,0 +1,85 @@
+# Copyright 2023 Google Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from dataclasses import dataclass
+import logging
+
+import pandas as pd
+from stats.config import Config
+from stats.data import StatVar
+from stats.data import StatVarGroup
+from stats.data import Triple
+from util.filehandler import FileHandler
+
+_CUSTOM_SV_ID_PREFIX = "custom/statvar_"
+_CUSTOM_GROUP_ID_PREFIX = "custom/g/group_"
+_ROOT_GROUP_ID = "dc/g/Root"
+
+
+class Nodes:
+
+  def __init__(self, config: Config) -> None:
+    self.config = config
+    # Dictionary of SVs from column name to SV
+    self.variables: dict[str, StatVar] = {}
+    # Dictionary of SVGs from SVG path to SVG
+    self.groups: dict[str, StatVarGroup] = {}
+
+  def variable(self, sv_column_name: str) -> StatVar:
+    if sv_column_name in self.variables:
+      return self.variables[sv_column_name]
+
+    var_cfg = self.config.variable(sv_column_name)
+    group = self.group(var_cfg.group_path)
+    group_id = group.id if group else _ROOT_GROUP_ID
+    self.variables[sv_column_name] = StatVar(
+        f"{_CUSTOM_SV_ID_PREFIX}{len(self.variables) + 1}",
+        var_cfg.name,
+        description=var_cfg.description,
+        nl_sentences=var_cfg.nl_sentences,
+        group_id=group_id,
+    )
+    return self.variables[sv_column_name]
+
+  def group(self, group_path: str) -> StatVarGroup | None:
```

Comment:
just thinking ahead, if we use column name as dcid, then the group path might to have the stat var name. for exampele:

"country_risk": "Floret/Country_Risk"

Reply:
Group paths don't have stat var names. In the above case, the group name will only be "Floret" and the group ID is generated by us (e.g. "custom/g/group_1")

Reply:
Oh, now i remember, there is a separate name field for the stat var.